scripts/check-monitoring-infrastructure.sh

```
@@ -0,0 +1,246 @@
+    fi
+}
+
+check_label() {
+    local resource=$1
+    local name=$2
+    local namespace=$3
+    local label=$4
+    
+    # Use jq to handle dotted label keys properly (e.g., app.kubernetes.io/component)
+    # JSONPath can't handle keys with dots/slashes correctly
+    local label_value=$(kubectl get "$resource" "$name" -n "$namespace" -o json 2>/dev/null | jq -r ".metadata.labels[\"$label\"]" 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)